### DISCUSSION TITLE
how to raise an HTTPException alongside a redirect response?

### DISCUSSION THREAD

**Discussion by asaah18** [fastapi/fastapi#4424] [Questions] [answered]
The use case at hand is as follow, I want to:

- Raise an `HTTPException` if a user is **not authorized** -to be specific, **authorization token not found or invalid**-.
- In addition, I want to redirect the client to the login end-point -which return an HTML response-.

I know how to do them individually, but not together/at the same time.

Any help is appreciated

**Comment by jgould22** [comment, marked as answer]
You can probably do this with something like though I have not tested it 
```python
raise HTTPException(status_code=302, detail="Not authorized", headers = {"Location": "your/login/route"} )
```

Where the 302 tells the browser to redirect and the location header tells it where to redirect

These links might also help 

https://fastapi.tiangolo.com/tutorial/handling-errors/#raise-an-httpexception-in-your-code

https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections

**Reply by asaah18** [reply]
Thank you, that worked out
